src/plugins/gs-plugin-external-apps.c

```diff
@@ -375,10 +388,19 @@ get_external_runtime_from_json (GsPlugin *plugin,
 
 	runtime = gs_plugin_cache_lookup (plugin, full_id);
 	if (runtime) {
-		g_debug ("Found cached '%s'", full_id);
 		gs_app_set_management_plugin (runtime,
 					      gs_plugin_get_name (plugin));
 
+		if (gs_flatpak_is_installed (priv->sys_flatpak, runtime, NULL,
```

Comment:
I think that this block of code and the one at the end of this function should be refactored together.

So basically you'd have
```
if (runtime) {
  g_debug("...");
} else {
  // create runtime here
}

// check installed state and set mgmt plugin
return runtime;
```